app/routes/demo/code-mirror.ts

```diff
@@ -1,3 +1,169 @@
 import Route from '@ember/routing/route';
 
-export default class DemoCodeMirrorRoute extends Route {}
+export default class DemoCodeMirrorRoute extends Route {
+  queryParams = {
+    allowMultipleSelections: {
+      replace: true,
+    },
+
+    autocompletion: {
+      replace: true,
+    },
+
+    bracketMatching: {
+      replace: true,
+    },
+
+    closeBrackets: {
+      replace: true,
+    },
+
+    collapseUnchanged: {
+      replace: true,
+    },
+
+    crosshairCursor: {
+      replace: true,
+    },
+
+    document: {
+      replace: true,
+    },
+
+    drawSelection: {
+      replace: true,
+    },
+
+    dropCursor: {
+      replace: true,
+    },
+
+    editable: {
+      replace: true,
+    },
+
+    filename: {
+      replace: true,
+    },
+
+    foldGutter: {
+      replace: true,
+    },
+
+    highlightActiveLine: {
+      replace: true,
+    },
+
+    highlightSelectionMatches: {
+      replace: true,
+    },
+
+    highlightSpecialChars: {
+      replace: true,
+    },
+
+    highlightTrailingWhitespace: {
+      replace: true,
+    },
+
+    highlightWhitespace: {
+      replace: true,
+    },
+
+    history: {
+      replace: true,
+    },
+
+    indentOnInput: {
+      replace: true,
+    },
+
+    indentUnit: {
+      replace: true,
+    },
+
+    indentWithTab: {
+      replace: true,
+    },
+
+    language: {
+      replace: true,
+    },
+
+    lineNumbers: {
+      replace: true,
+    },
+
+    lineSeparator: {
+      replace: true,
+    },
+
+    lineWrapping: {
+      replace: true,
+    },
+
+    mergeControls: {
+      replace: true,
+    },
+
+    originalDocument: {
+      replace: true,
+    },
+
+    outline: {
+      replace: true,
+    },
+
+    placeholder: {
+      replace: true,
+    },
+
+    preserveHistory: {
+      replace: true,
+    },
+
+    readOnly: {
+      replace: true,
+    },
+
+    rectangularSelection: {
+      replace: true,
+    },
+
+    scrollPastEnd: {
+      replace: true,
+    },
+
+    selectedDocumentIndex: {
+      replace: true,
+    },
+
+    selectedIndentUnitIndex: {
+      replace: true,
+    },
+
+    selectedLineSeparatorIndex: {
+      replace: true,
+    },
+
+    selectedTabSizeIndex: {
+      replace: true,
+    },
+
+    selectedThemeIndex: {
+      replace: true,
+    },
+
+    syntaxHighlighting: {
+      replace: true,
+    },
+
+    tabSize: {
+      replace: true,
+    },
+
+    theme: {
+      replace: true,
+    },
+  };
```

Comment:
Good idea

Reply:
Doesn't work well in TypeScript though, as `acc` now implicitly has `any` type

Reply:
Wow, true human machine symbiosis going on here

Reply:
I was impressed the most about how it learned something from me and adjusted itself for future better-ness

> Learnings added:
> Learning: When refactoring TypeScript code using reduce to build an object, explicitly define the accumulator's type to avoid implicit 'any' types.